main.go

```diff
@@ -130,12 +127,58 @@ func Upload(c *gin.Context) {
 		if err != nil {
 			log.Fatalf("failed to upload archive to AWS-Glacier, %v", err)
 		}
-		c.String(http.StatusOK, "File %s uploaded successfully with id %s", filename, *result.ArchiveId)
+
+		// save data to db
+		archive_data := util.Archive{
+			Vault_file_id: *result.ArchiveId,
+			Library_id:    0,
+			User_id:       0,
+			File_name:     filename,
+			Upload_date:   "now",
+			File_size:     0,
```

Comment:
These data needs to be legit already as we talked about and make sure it's stored in DB.